features/admin.application-templates.v1/pages/application-template.tsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2023-2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+    AppConstants,
+    history
+} from "@wso2is/admin.core.v1";
+import { ExtensionTemplateListInterface } from "@wso2is/admin.template-core.v1/models/templates";
+import { TestableComponentInterface } from "@wso2is/core/models";
+import { PageLayout } from "@wso2is/react-components";
+import React, { FunctionComponent, ReactElement, useState } from "react";
+import { useTranslation } from "react-i18next";
+import ApplicationCreationAdapter from "../components/application-creation-adapter";
+import ApplicationTemplateGrid from "../components/application-templates-grid";
+import ApplicationTemplateMetadataProvider from
+    "../provider/application-template-metadata-provider";
+import ApplicationTemplateProvider from "../provider/application-template-provider";
+
+/**
+ * Props for the Applications templates page.
+ */
+type ApplicationTemplateSelectPageInterface = TestableComponentInterface;
+
+/**
+ * Choose the application template from this page.
+ *
+ * @param props - Props injected to the component.
+ *
+ * @returns Application template select page.
+ */
+const ApplicationTemplateSelectPage: FunctionComponent<ApplicationTemplateSelectPageInterface> = ({
+    "data-testid": testId = "application-templates"
+}: ApplicationTemplateSelectPageInterface): ReactElement => {
+
+    const { t } = useTranslation();
+
+    const [ showWizard, setShowWizard ] = useState<boolean>(false);
+    const [ selectedTemplate, setSelectedTemplate ] = useState<ExtensionTemplateListInterface>(null);
+
+    /**
+     * Handles back button click.
+     */
+    const handleBackButtonClick = (): void => {
+        history.push(AppConstants.getPaths().get("APPLICATIONS"));
+    };
+
+    return (
+        <PageLayout
+            pageTitle="Register New Application"
```

Comment:
Don't we need to i18n this?

Reply:
This is an existing code block. https://github.com/wso2/identity-apps/blob/05605b9367987c95ce06add63ce0e95494a90b2d/features/admin.applications.v1/pages/application-template.tsx#L435-L436